WordPress/WordPressTodayWidget/TodayViewController.swift

```diff
@@ -44,50 +60,79 @@ class TodayViewController: UIViewController, NCWidgetProviding {
         self.visitorCount = userDefaults.stringForKey(WPStatsTodayWidgetUserDefaultsVisitorCountKey) ?? "0"
         self.viewCount = userDefaults.stringForKey(WPStatsTodayWidgetUserDefaultsViewCountKey) ?? "0"
         
-        self.siteNameLabel?.text = self.siteName
-        self.visitorsCountLabel?.text = self.visitorCount
-        self.viewsCountLabel?.text = self.viewCount
+        self.siteNameLabel.text = self.siteName
+        self.visitorsCountLabel.text = self.visitorCount
+        self.viewsCountLabel.text = self.viewCount
     }
     
     @IBAction func launchContainingApp() {
-        self.extensionContext!.openURL(NSURL(string: "\(WPCOM_SCHEME)://viewstats?siteId=\(siteId!)")!, completionHandler: nil)
+        if let unwrappedSiteID = siteID {
+            self.extensionContext!.openURL(NSURL(string: "\(WPCOM_SCHEME)://viewstats?siteId=\(unwrappedSiteID)")!, completionHandler: nil)
+        } else {
+            self.extensionContext!.openURL(NSURL(string: "\(WPCOM_SCHEME)://")!, completionHandler: nil)
+        }
     }
     
-    func widgetPerformUpdateWithCompletionHandler(completionHandler: ((NCUpdateResult) -> Void)) {
-        // Perform any setup necessary in order to update the view.
-        
-        // If an error is encoutered, use NCUpdateResult.Failed
-        // If there's no update required, use NCUpdateResult.NoData
-        // If there's an update, use NCUpdateResult.NewData
+    func getOAuth2Token() -> String? {
+        let oauth2Token = try? SFHFKeychainUtils.getPasswordForUsername(WPStatsTodayWidgetOAuth2TokenKeychainUsername, andServiceName: WPStatsTodayWidgetOAuth2TokenKeychainServiceName, accessGroup: WPStatsTodayWidgetOAuth2TokenKeychainAccessGroup)
         
+        return oauth2Token as String?
+    }
+    
+    func updateUIBasedOnWidgetConfiguration() {
+        siteNameLabel.hidden = !isConfigured
+        countContainerView.hidden = !isConfigured
```

Comment:
Interesting thing here.  I see there is a height constraint set on `countContainerView` in the storyboard.  When `hidden = true` the UIStackView will set another height constraint == 0 to collapse the view. I'm not seeing any warnings of breaking constraints so the priorities are probably playing nice.  Still, it might not be a best practice to set height constraints on the immediate subviews of a UIStackView.